backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java

```diff
@@ -163,109 +171,115 @@ public TestResultUpload processResultCSV(InputStream csvStream) {
-    var testResultDateIndex = headers.indexOf(TEST_RESULT_DATE_COLUMN_NAME);
-    var orderTestDateIndex = headers.indexOf(ORDER_TEST_DATE_COLUMN_NAME);
-    var specimenCollectionDateIndex = headers.indexOf(SPECIMEN_COLLECTION_DATE_COLUMN_NAME);
-    var specimenReceivedDateIndex = headers.indexOf(TESTING_LAB_SPECIMEN_RECEIVED_DATE_COLUMN_NAME);
-    var dateResultReleasedIndex = headers.indexOf(DATE_RESULT_RELEASED_COLUMN_NAME);
-
-    var testResultDate = valueAtRowIndex(testResultDateIndex, row);
-    var orderTestDate = valueAtRowIndex(orderTestDateIndex, row);
-    var specimenCollectionDate = valueAtRowIndex(specimenCollectionDateIndex, row);
-    var specimenReceivedDate = valueAtRowIndex(specimenReceivedDateIndex, row);
-    var dateResultReleased = valueAtRowIndex(dateResultReleasedIndex, row);
-
-    var testingLabAddr = getTestingLabAddress(row, headers);
-    var providerAddr = getOrderingFacilityAddress(row, headers);
-
-    testResultDate =
-        convertToZonedDateTime(testResultDate, resultsUploaderCachingService, testingLabAddr)
-            .toOffsetDateTime()
-            .toString();
-    orderTestDate =
-        convertToZonedDateTime(orderTestDate, resultsUploaderCachingService, providerAddr)
-            .toOffsetDateTime()
-            .toString();
-
-    specimenCollectionDate =
-        StringUtils.isNotBlank(specimenCollectionDate)
+  private Map<String, String> transformCsvRow(Map<String, String> row) {
+    var updatedSpecimenType =
+        modifyRowSpecimenNameToSNOMED(row.get(SPECIMEN_TYPE_COLUMN_NAME).toLowerCase());
+
+    var testingLabAddr =
+        new StreetAddress(
+            row.get("testing_lab_street"),
+            row.get("testing_lab_street2"),
+            row.get("testing_lab_city"),
+            row.get("testing_lab_state"),
+            row.get("testing_lab_zip_code"),
+            null);
+    var providerAddr =
+        new StreetAddress(
+            row.get("ordering_provider_street"),
+            row.get("ordering_provider_street2"),
+            row.get("ordering_provider_city"),
+            row.get("ordering_provider_state"),
+            row.get("ordering_provider_zip_code"),
+            null);
+
+    var testResultDate =
+        convertToZonedDateTime(
+            row.get(TEST_RESULT_DATE_COLUMN_NAME), resultsUploaderCachingService, testingLabAddr);
+
+    var orderTestDate =
+        convertToZonedDateTime(
+            row.get(ORDER_TEST_DATE_COLUMN_NAME), resultsUploaderCachingService, providerAddr);
+
+    var specimenCollectionDate =
+        StringUtils.isNotBlank(row.get(SPECIMEN_COLLECTION_DATE_COLUMN_NAME))
             ? convertToZonedDateTime(
-                    specimenCollectionDate, resultsUploaderCachingService, providerAddr)
-                .toOffsetDateTime()
-                .toString()
+                row.get(SPECIMEN_COLLECTION_DATE_COLUMN_NAME),
+                resultsUploaderCachingService,
+                providerAddr)
```

Comment:
Does this need to be changed to the testingLabAddr

Reply:
From #5987, a blank `specimen_collection_date` field is populated by `order_test_date` so based on that I'm assuming we should use the ordering provider address for both of those fields